docs/docs/workflows/upgrade.md

```diff
@@ -1,44 +1,50 @@
+If your current Kubernetes version is dropped from the next CLI version, use your current CLI to upgrade to a newer Kubernetes version first.
+Each CLI supports at least three Kubernetes versions.
+To learn which Kubernetes versions are supported by a particular CLI run [constellation config kubernetes-versions](../reference/cli.md#constellation-config-kubernetes-versions).
+
 
 ## Migrate the configuration
 
 The Constellation configuration file is located in the file `constellation-conf.yaml` in your workspace.
 Refer to the [migration reference](../reference/config-migration.md) to check if you need to update fields in your configuration file.
 
-## Plan the upgrade
+## Check for upgrades
 
-If you don't already know the image you want to upgrade to, use the `upgrade plan` command to pull a list of available updates.
+To learn which versions the current CLI can upgrade to and what's installed in your cluster use this command to see available upgrades:
 
 ```bash
-constellation upgrade plan
+constellation upgrade check
 ```
 
-The command lets you interactively choose from a list of available updates and prepares your Constellation config file for the next step.
-
-To use the command in scripts, use the `--file` flag to compile the available options into a YAML file.
-You can then set the chosen upgrade option in your Constellation config file.
-
-:::caution
-
-`constellation upgrade plan` only works for official Edgeless release images.
-If your cluster is using a custom image, the Constellation CLI will fail to find compatible images.
-However, you may still use the `upgrade execute` command by manually selecting a compatible image and setting it in your config file.
-
-:::
+You can either enter the reported target versions into your config manually or run the above command with the `--write-config` flag.
+When using this flag the `kubernetesVersion`, `image` and `microserviceVersion` fields are overwritten with the smallest available upgrade.
 
-## Execute the upgrade
+## Apply the upgrade
 
-Once your config file has been prepared with the new image and measurements, use the `upgrade execute` command to initiate the upgrade.
+Once you updated your config with the versions you want in your cluster, you can trigger the upgrade with this command:
 
 ```bash
-constellation upgrade execute
+constellation upgrade apply
 ```
 
-After the command has finished, the cluster will automatically replace old nodes using a rolling update strategy to ensure no downtime of the control or data plane.
+Microservice upgrades will be finished within a few minutes, depending on the cluster size.
+If you are interested, you can monitor pods restarting in the `kube-system` namespace with your tool of choice.
+Image and Kubernetes upgrades will take longer.
+This is because for each node in your cluster, a new node has to be created and joined.
+The process usually takes ~6 minutes per node.
```

Comment:
```suggestion
If you are interested, you can monitor pods restarting in the `kube-system` namespace with your tool of choice.

Image and Kubernetes upgrades take longer.
For each node in your cluster, a new node has to be created and joined.
The process usually takes about 6 minutes per node.
```
I'd start a new paragraph (and simplify the wording a bit).

6 minutes sound like a very accurate timing. Is this really true, independent of CSP or node type? (If not, maybe replace with sth like "about five minutes" or "up to ten minutes"?)